python-avd/schema_tools/compile_templates.py

```diff
@@ -33,39 +33,79 @@ def compile_eos_designs_templates() -> None:
     templar.compile_templates_in_paths(precompiled_templates_path=EOS_DESIGNS_JINJA2_PRECOMPILED_TEMPLATE_PATH, searchpaths=[EOS_DESIGNS_JINJA2_TEMPLATE_PATH])
 
 
-def check_eos_designs_templates() -> None:
-    """Verify if eos_designs need to be recompiled when running from source."""
+def check_eos_designs_templates() -> bool:
+    """
+    Verify if eos_designs need to be recompiled when running from source.
+
+    Returns:
+    --------
+    bool:
+        True if any templates changed, False otherwise
+    """
     if not RUNNING_FROM_SRC:
-        return
+        return False
 
     LOGGER.info("pyavd running from source detected, checking eos_designs templates for any changes...")
     dir_path = EOS_DESIGNS_JINJA2_TEMPLATE_PATH
 
-    changed, new_hash = changed_hash(dir_path)
-    if changed:
-        LOGGER.info("Recompiling eos_designs templates...")
-        compile_eos_designs_templates()
-        with (dir_path / ".hash").open("w") as fd:
-            fd.write(new_hash)
+    changed, _ = changed_hash(dir_path)
+    return changed
 
 
-def check_eos_cli_config_gen_templates() -> None:
-    """Verify if eos_cli_config_gen need to be recompiled when running from source."""
+def recompile_eos_designs_templates() -> None:
+    """Recompile eos_designs templates."""
+    LOGGER.info("Recompiling eos_designs templates...")
+    dir_path = EOS_DESIGNS_JINJA2_TEMPLATE_PATH
+    compile_eos_designs_templates()
+    new_hash = hash_dir(dir_path)
+    with (dir_path / ".hash").open("w") as fd:
+        fd.write(new_hash)
+
+
+def check_eos_cli_config_gen_templates() -> bool:
+    """
+    Verify if eos_cli_config_gen need to be recompiled when running from source.
+
+    Returns:
+    --------
+    bool:
+        True if any templates changed, False otherwise
+    """
     if not RUNNING_FROM_SRC:
-        return
+        return False
 
     LOGGER.info("pyavd running from source detected, checking eos_cli_config_gen templates for any changes...")
     dir_path = EOS_CLI_CONFIG_GEN_JINJA2_TEMPLATE_PATH
 
     changed, new_hash = changed_hash(dir_path)
```

Comment:
```suggestion
    changed, _ = changed_hash(dir_path)
```

Reply:
We are generally keeping the name for readability so I suggest `_new_hash` to not trigger linting issues.